### PR TITLE
SPEC: discovery clarifications

### PR DESCRIPTION
**Pull request by jonboulle**
- codify/define "discovery URL" + template
- remove misleading `?torrent` tag
- remove ACE references, instead talk about discovery clients
- discuss behaviour in case of multiple `ac-discovery` tags

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -437,17 +441,19 @@ curl $(echo "$urltmpl" | sed -e "s/{name}/$appname/" -e "s/{version}/$version/"
...
 ```
-sig: https://storage.example.com/linux/amd64/reduce-worker-1.0.0.aci.asc
-aci: https://storage.example.com/linux/amd64/reduce-worker-1.0.0.aci
-keys: https://example.com/pubkeys.gpg
+Signature: 	https://storage.example.com/linux/amd64/reduce-worker-1.0.0.aci.asc
+ACI: 		https://storage.example.com/linux/amd64/reduce-worker-1.0.0.aci
+Keys: 		https://example.com/pubkeys.gpg
 ```
 
-This mechanism is only used for discovery of contents URLs.
```
Comment (jonboulle) [inline SPEC.md]: I didn't understand this comment

**Comment by jonboulle** [comment]
Fixes #283

File: SPEC.md [hunk 2]
```diff
@@ -437,17 +441,19 @@ curl $(echo "$urltmpl" | sed -e "s/{name}/$appname/" -e "s/{version}/$version/"
...
-
-If the first attempt at fetching the discovery URL returns a status code other than `200 OK`, `3xx`, or does not contain any `ac-discovery` meta tags then the next higher path in the name should be tried.
+If the first attempt at fetching the initial discovery URL returns a status code other than `200 OK`, `3xx`, or does not contain any `ac-discovery` meta tags then the next higher path in the name should be tried.
```
Comment (eyakubovich) [inline SPEC.md]: I'm not sure if 5xx errors should cause the discover to bubble up. It's usually a temporary error indicative of the server malfunction and should be retried at a later point in time.

Comment (jonboulle) [inline SPEC.md]: I am not really sure why it is framed this way rather than the inverse, e.g. calling out 4xx codes. @philips ?

File: SPEC.md [hunk 3]
```diff
@@ -406,30 +406,34 @@ Simple discovery does not provide a way to discover Public Keys.
 
 ### Meta Discovery
 
-If simple discovery fails, then we use HTTPS+HTML meta tags to resolve an app name to a downloadable URL.
-For example, if the ACE is looking for `example.com/reduce-worker` it will request:
+If simple discovery fails, then we use HTTPS+HTML meta tags retrieved from a "discovery URL" to resolve an app name to downloadable URLs.
+The template for the discovery URL is
```
Comment (philips) [inline SPEC.md]: Should this be on a newline with a colon "discovery URL is:"

File: SPEC.md (hunk 2 above)
Comment (philips) [inline SPEC.md]: doing 4xx makes more sense to me.

**Comment by philips** [comment]
lgtm

File: SPEC.md (hunk 3 above)
Comment (jonboulle) [inline SPEC.md]: ok

File: SPEC.md (hunk 2 above)
Comment (jonboulle) [inline SPEC.md]: https://github.com/jonboulle/spec/commit/1f28c3a77b5c0a97703e38159aabc75a30155618

**Pull request merged**